src/Util.hpp

```diff
@@ -405,11 +406,33 @@ starts_with(nonstd::string_view string, nonstd::string_view prefix)
   return string.starts_with(prefix);
 }
 
+struct KeyAndValueView
+{
+  bool
+  hasKeyAndValue() const
+  {
+    return !key.empty() && !value.empty();
+  }
+  bool
+  hasBeenSplit() const
+  {
+    return !key.empty() || !value.empty();
+  }
+
+  nonstd::string_view key, value;
+};
+// Splits strings like "key=value" into key and value.
+// If string does not contain splitChar, it will return empty values.
+// Note: this works on string_view. The original string must remain available
+// for the result to be usable.
+KeyAndValueView splitKeyAndValue(nonstd::string_view string,
+                                 const char splitChar = '=');
```

Comment:
That first signature is actually exactly what I had in my first iteration. But I quickly decided that while correct, it's way nicer to reason about key and value instead of first and second on the call sites. So that's why I have this struct.
Next I got rid of the optional, because while it feels natural to write the function that way, it's kind of strange to work with. You would have to write `if(key_and_value && (key_and_value.key == "-short" || key_and_value.key == "--long")`.

So both 'optimizations' are for make the caller code simpler. It's used a lot and I wanted it as readable as possible.